text/0000-bee-model-crate/0000-bee-model-crate.md

```diff
@@ -0,0 +1,334 @@
+A bundle then represents the collection of these transactions. It should be noted, that each transaction of a bundle will be sent separately through the network.
+
+Each transaction is identified by its hash, the transaction hash. Transactions should be final and therefore not modifiable after construction.
+The same applies to bundles. Bundles are final, and therefore not modifiable after construction.
+
+Transactions can be built from Transaction Builders.
+Bundles can be built from Bundle Builders. Both builder objects can be manipulated as desired.
+It should be noted, Bundle Builders don't process Transactions (as transactions are final), instead, they process Transaction Builders.
+
+## Exposed Interface
+
+### Transaction
+
+ A transaction consists of following fields:
+
+```rust
+struct Transaction {
+
+    signature_fragments: String,
+    address: String,
+    value: i64,
+    obsolete_tag: String,
+    timestamp: i64,
+    current_index: usize,
+    last_index: usize,
+    bundle_hash: String,
+    trunk: String,
+    branch: String,
+    nonce: String,
+    tag: String,
+    attachment_timestamp: i64,
+    attachment_timestamp_lower_bound: i64,
+    attachment_timestamp_upper_bound: i64,
+
+}
+```
+
+As mentioned, Transactions are final. Transaction fields therefore should be only accessible by getter functions.
+Besides that, a transaction can only be created from a Transaction Builder, or from a constructor function which expects encoded bytes as received e.g. from a network socket.
+
+```rust
+impl Transaction {
+    
+    pub fn from_bytes(bytes: &Vec<u8>) -> Result<Transaction, TransactionBuilderValidationError> {
+        unimplemented!()
+    }
+
+    pub fn signature_fragments(&self) -> &String {
+        &self.signature_fragments
+    }
+
+    pub fn address(&self) -> &String {
+        &self.address
+    }
+
+    pub fn value(&self) -> &i64 {
+        &self.value
+    }
+
+    pub fn obsolete_tag(&self) -> &String {
+        &self.obsolete_tag
+    }
+
+    pub fn timestamp(&self) -> &i64 {
+        &self.timestamp
+    }
+
+    pub fn current_index(&self) -> &usize {
+        &self.current_index
+    }
+
+    pub fn last_index(&self) -> &usize {
+        &self.last_index
+    }
+
+    pub fn bundle_hash(&self) -> &String {
+        &self.bundle_hash
+    }
+
+    pub fn trunk(&self) -> &String {
+        &self.trunk
+    }
+
+    pub fn branch(&self) -> &String {
+        &self.branch
+    }
+
+    pub fn nonce(&self) -> &String {
+        &self.nonce
+    }
+
+    pub fn tag(&self) -> &String {
+        &self.tag
+    }
+
+    pub fn attachment_timestamp(&self) -> &i64 {
+        &self.attachment_timestamp
+    }
+
+    pub fn attachment_timestamp_lower_bound(&self) -> &i64 {
+        &self.attachment_timestamp_lower_bound
+    }
+
+    pub fn attachment_timestamp_upper_bound(&self) -> &i64 {
+        &self.attachment_timestamp_upper_bound
+    }
+
+}
+```
```

Comment:
I think you can safely remove most of the getter functions above to demonstrate what you mean. :) Let's mention that the fields should be left private and immutable.